### PR TITLE
Esha/role assignments

### PR DESCRIPTION
**Pull request by eshabansiya**
### 🦋 Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

Query: assignRoles() - changes the roleID of a userID 
Updated queries: 
- fetchParticipants: now returns name and ID
- fetchRoles: now returns roles and ID
- --> did that so that in role-assignment, the ID can be used to update IDs while names are used to be displayed in the dropdown

role-selection/page.tsx:
- was not able to test pulling user_group_id from profile context
- dropdowns to assing participant's roles
- button to add more participants to assign
- In-line styling to get the buttons next to each other... will change when more styling....
- Save button to save all assigned participants

Note: May not be consistent with new PR comments

### 🦧 Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

Screenshot 2025-11-02 at 3 57 18 PM

Screenshot 2025-11-02 at 3 57 30 PM

(i dont have screenshots of values bc i cant test :( , but it would work)

## 🐸 How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Please check if the user_group_id pull works :(
check supabase if things update 

CC: @me-liu

### CODE REVIEW

File: app/sessions/role-selection/page.tsx [hunk 1]
```diff
@@ -1,25 +1,39 @@
...
+    { participant: null, role: null },
+  ]);
 
   useEffect(() => {
+    if (loading || !profile?.user_group_id) return;
+    const userGroupId = profile.user_group_id;
+
     async function loadData() {
       try {
         //idk where to get the ids from... so i just used from the supabase
```
Comment (me-liu) [inline app/sessions/role-selection/page.tsx]: delete comment thankss

File: app/sessions/role-selection/page.tsx [hunk 2]
```diff
@@ -32,28 +46,89 @@ export default function RoleSelectionPage() {
...
+          key={index}
+          style={{
+            display: "flex",
+            alignItems: "center",
+            marginBottom: "1rem",
+          }}
+        >
+          <InputDropdown
+            label={`Name ${index + 1}`}
+            options={new Map(participants.map(p => [p.id, p.name]))}
```
Comment (me-liu) [inline app/sessions/role-selection/page.tsx]: Can you make it so that each inputDropdown only will have participants that aren't selected beforehand? This will ensure that participants don't have multiple roles. Hopefully this shouldn't be too complicated--just some filtering here

**Review by me-liu — COMMENTED** [review]
Looks solid! Just some error checking + delete.

File: app/sessions/role-selection/page.tsx [hunk 3]
```diff
@@ -1,59 +1,150 @@
...
-          label="Select Participant"
-          options={new Set(participants)}
-          placeholder="Choose a participant"
-          onChange={setSelectedParticipant}
-        />
-      </>
+      {roleSelection.map((pair, index) => {
+        const selectedParticipants = roleSelection
+          .map(p => p.participant)
+          .filter((p, i) => p && i !== index);
```
Comment (me-liu) [inline app/sessions/role-selection/page.tsx]: actually can you ensure that you don't include your own userId in here too, just so you can't play and also be facilitator

**Review by me-liu — COMMENTED** [review]
okayyy fire!! i tested profileContext and things work perfectly ty ty just another nit and then this is all good

**Review by me-liu — APPROVED** [review]
